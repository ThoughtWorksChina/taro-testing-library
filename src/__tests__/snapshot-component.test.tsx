import Taro, {Component} from '@tarojs/taro';
import { render } from '../index';
import {StandardProps} from "@tarojs/components/types/common";
import {Text} from "@tarojs/components";

interface CounterProps extends StandardProps {
  initial?: number;
}

interface CounterState {
  count: number;
}

class Counter extends Component<CounterProps, CounterState> {

  static defaultProps = {
    initial: 1
  }

  static externalClasses = ['custom-class'];

  constructor(props: CounterProps) {
    super(props);
    this.state = {
      count: props.initial as number
    }
  }

  render() {
    const { count } = this.state
    return (
      <Text
        onClick={() => {this.setState({
          count: count + 1
        })}}
        className="number custom-class"
      >{count}</Text>
    )
  }
}

describe('snapshot test', () => {
  it('should pass snapshot', () => {
    const { baseElement } = render(<Counter />);
    expect(baseElement).toMatchSnapshot()
  });
});
