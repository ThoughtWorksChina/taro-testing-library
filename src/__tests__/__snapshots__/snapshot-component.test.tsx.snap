// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test should pass snapshot 1`] = `
<div>
  <span
    class="taro-text number custom-class"
  >
    1
  </span>
</div>
`;
